baby-liminal-extension/src/lib.rs

```diff
@@ -0,0 +1,115 @@
+#![cfg_attr(not(feature = "std"), no_std)]
+
+#[cfg(feature = "ink")]
+pub mod ink;
+
+#[cfg(feature = "substrate")]
+pub mod substrate;
+
+#[cfg(feature = "substrate")]
+pub mod executor;
+
+#[cfg(feature = "ink")]
+use ::ink::prelude::vec::Vec;
+#[cfg(feature = "substrate")]
+use obce::substrate::sp_std::vec::Vec;
+use scale::{Decode, Encode};
+#[cfg(feature = "std")]
+use scale_info::TypeInfo;
+
+// `pallet_baby_liminal::store_key` errors
+const BABY_LIMINAL_STORE_KEY_ERROR: u32 = 10_000;
+pub const BABY_LIMINAL_STORE_KEY_TOO_LONG_KEY: u32 = BABY_LIMINAL_STORE_KEY_ERROR + 1;
+pub const BABY_LIMINAL_STORE_KEY_IDENTIFIER_IN_USE: u32 = BABY_LIMINAL_STORE_KEY_ERROR + 2;
+pub const BABY_LIMINAL_STORE_KEY_ERROR_UNKNOWN: u32 = BABY_LIMINAL_STORE_KEY_ERROR + 3;
+
+// `pallet_baby_liminal::verify` errors
+const BABY_LIMINAL_VERIFY_ERROR: u32 = 11_000;
+pub const BABY_LIMINAL_VERIFY_DESERIALIZING_PROOF_FAIL: u32 = BABY_LIMINAL_VERIFY_ERROR + 1;
+pub const BABY_LIMINAL_VERIFY_DESERIALIZING_INPUT_FAIL: u32 = BABY_LIMINAL_VERIFY_ERROR + 2;
+pub const BABY_LIMINAL_VERIFY_UNKNOWN_IDENTIFIER: u32 = BABY_LIMINAL_VERIFY_ERROR + 3;
+pub const BABY_LIMINAL_VERIFY_DESERIALIZING_KEY_FAIL: u32 = BABY_LIMINAL_VERIFY_ERROR + 4;
+pub const BABY_LIMINAL_VERIFY_VERIFICATION_FAIL: u32 = BABY_LIMINAL_VERIFY_ERROR + 5;
+pub const BABY_LIMINAL_VERIFY_INCORRECT_PROOF: u32 = BABY_LIMINAL_VERIFY_ERROR + 6;
+pub const BABY_LIMINAL_VERIFY_ERROR_UNKNOWN: u32 = BABY_LIMINAL_VERIFY_ERROR + 7;
+
+/// Chain extension errors enumeration.
+///
+/// All inner variants are convertible to [`RetVal`]
+/// via [`TryFrom`] impl.
+///
+/// To ensure that [`RetVal`] is returned when possible in implementation
+/// its methods should be marked with `#[obce(ret_val)]`.
+///
+/// [`RetVal`]: obce::substrate::pallet_contracts::chain_extension::RetVal
+#[obce::error]
```

Comment:
out of scope of this PR, but could we make this macro ensure two things:
1. no ret val is duplicated
2. every variant has `#[obce(ret_val` attribute

?